lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -53,12 +53,17 @@ def validate_client
       def validate_scopes
         return true unless scope.present?
         Helpers::ScopeChecker.valid?(
-          scope,
-          server.scopes,
-          client.application.scopes
+          scope_str: scope,
+          server_scopes: server.scopes,
+          app_scopes: client.application.scopes,
+          grant_type: grant_type
         )
       end
 
+      def grant_type
+        response_type == 'code' ? AUTHORIZATION_CODE : IMPLICIT
+      end
```

Comment:
naive question: this looks like pre-auth only happens for auth code and implicit grant flows - is that correct?